### ISSUE TITLE
inconsistency in the notation / term epoch 

### ISSUE THREAD

**Issue by iamyihwa** [zalandoresearch/flair#89]
Hello, 
I am training language model, 
Before testing with whole text, I was running it with smaller text. 

From my understanding epoch vs. batch vs. minibatch is: (From a post in [stackexchange](https://stats.stackexchange.com/questions/117919/what-are-the-differences-between-epoch-batch-and-minibatch))
![image](https://user-images.githubusercontent.com/12494156/44517496-2dfc9c80-a6c8-11e8-9638-a32ea278946f.png)
 
However when train the language model with following parameters, 
```
# train your language model
trainer = LanguageModelTrainer(language_model, corpus)

trainer.train('resources/taggers/language_model_es_forward',
              sequence_length=250,
              mini_batch_size=100,
              max_epochs=5)

```
I get following output. 
Currently there are 51 input files. ( I see that the mini_batch_size is larger than the number of the training files. So this might be a problem) 
However, in the output, the epoch number doesn't change ..  only it says end of split (1/ 51) and it finishes after (5 / 51) ..
I wonder if it is due to different use of terminology? 
Or in this case it only goes through 5 files and stops ? 
If I want to go through my entire dataset 100 times for example, I have to do 100 * number_of_train_files as max_epochs? 
 
(The Screenshot might be confusing, but the training finished after 5 / 51.. ) 
![image](https://user-images.githubusercontent.com/12494156/44517425-f130a580-a6c7-11e8-927d-697d22d27304.png)

The original dataset that I have (wikidump es)  are currently in 2300 files (each file about 1MB). 

I am intending to put about 5 files together and make it into validation set, and put about 5 files together and make it into test set.  

The rest of the files (about 2290 files), I will use to train the model. 

If I want to pass multiple times what value should I use for epochs? 

What are the good number of passes of data, when you trained your language model for English and German models?

**Comment by iamyihwa** [comment]
Thanks @alanakbik for the clarification. 
I will do as you suggest! 

At the moment I am seeing the loss not decreasing so quick, and ppl also got stuck somehow .. 
Is there anything I should be doing? 
![image](https://user-images.githubusercontent.com/12494156/44532925-95c8dc80-a6f4-11e8-9df5-901b919a30c0.png)

**Comment by iamyihwa** [comment]
@alanakbik Yes I have tried it with the larger data size, (also increased the hidden neuron numbers, just in case..)  and it already seems better! 

![image](https://user-images.githubusercontent.com/12494156/44703495-1d696f00-aa98-11e8-957c-3b5e38763789.png)


Thanks @alanakbik for the suggestion, I will try with patience = 100! 

BTW, does the language model require specific input ? I have used each one sentence input. In addition to that do I need to separate each token or do any normalization?
